src/sendCommercialMetrics.ts

```diff
@@ -1,72 +1,186 @@
 import { log } from '@guardian/libs';
 import { EventTimer } from './EventTimer';
 
-type Metrics = {
+type Metric = {
 	name: string;
 	value: number;
 };
 
-type Properties = {
+type Property = {
 	name: string;
-	value: string;
+	value: string | number;
 };
 
-type CommercialMetrics = {
-	browser_id?: string;
-	page_view_id: string;
-	platform: string;
-	metrics: readonly Metrics[];
-	properties: readonly Properties[];
+type TimedEvent = {
+	name: string;
+	ts: number;
+};
+
+type EventProperties = {
+	type?: ConnectionType;
+	downlink?: number;
+	effectiveType?: string;
+};
+
+type CommercialMetricsPayload = {
+	page_view_id: string | null;
+	browser_id?: string | null;
+	platform: 'NEXT_GEN' | null;
+	metrics: readonly Metric[] | null;
+	properties: readonly Property[] | null;
```

Comment:
Maybe introducing `null` here is just confusing. Either it’s `undefined` or it’s the expected type?

```suggestion
	page_view_id?: string;
	browser_id?: string;
	platform?: 'NEXT_GEN';
	metrics?: readonly Metric[];
	properties?: readonly Property[];
```

Reply:
Thank you - good suggestion 👍🏻